src/main/java/io/molnarsandor/trelloclone/collaborator/model/CollaboratorEntity.java

```diff
@@ -0,0 +1,32 @@
+package io.molnarsandor.trelloclone.collaborator.model;
+
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import io.molnarsandor.trelloclone.project.model.ProjectEntity;
+import io.molnarsandor.trelloclone.util.EntitySuperClass;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+import javax.validation.constraints.Email;
+
+@Entity
+@Table(name = "collaborator")
+@NoArgsConstructor
+@Data
+@EqualsAndHashCode(callSuper = false)
+public class CollaboratorEntity extends EntitySuperClass {
+
+    @Email(message = "Needs to be a valid email")
+    @Column(nullable = false)
+    private String email;
+    @Column(nullable = false)
+    private String projectIdentifier;
```

Comment:
Úgy látom használod a bean validációs annotációkat is, ugye itt kicsit fentebb az `@Email` annotációt, ezeket szeretem én is használni az entitásokon, viszont ami kicsit furcsább, hogy van két meződ is aminek a `@Column` annotációjában a `nullable`-t `false`-ra állítottad. Ebben az esetben ha a Hibernate-re bízod az adatbázis séma generálását, ami egy kisebb projektben, demo-ban, PoC projektben nagyon jó, mert gyorsan tudsz vele dolgozni, viszont nagyobb appoknál azt mondanám, hogy szigorúan tilos azt használni, erre inkább egy Flyway-t vagy egy Liquibase-t ajánlanék (második támogat több féle megközelítést a DB felépítéséhez, XML, JSON, YAML, a kettő közül nekem emiatt jobban tetszik a Liquibase :)

Úgy vélem egy fentebbi rétegben validálod, hogy ez a mező ne legyen null vagy üres, ugye itt lehet játszani a `@NotNull` és a `@NotBlank` annotációkkal. Arra figyelj, hogy a `@Column` annotáció nullable mezője nem fogja neked validálni, hogy ez az érték ne legyen null a bean validációs API-val, hanem csak a fentebb említett két másik annotáció.

Nálam az a preferált, ha a perzisztencia és bármilyen külső rétegben ahonnan adat érkezik be, ott validáljuk ezeket, mert ha az adatbázisig megy el és ott pukkan el akkor az drágább, mikor már hamarabb is megfoghattad volna ezt a problémát.